### PR TITLE
Fix Overflow Icon Color Change on Config Change (Orientation Change)

### LINKED ISSUE #11017: [Bug] Menu icons turn White after showing a new question with `Automatic display answer`
###### Reproduction Steps

1. Have `Automatic display answer` enabled
2. Start reviewing cards
3. Tap the menu items button

https://user-images.githubusercontent.com/69634269/164553626-e4d5f97c-5fea-489d-a588-45757127f930.mp4

###### Expected Result

The icons should keep their color

###### Actual Result

The icons turns white after a **new question** is shown (better seen on night themes)

###### Debug info
2.16alpha61: 22c5da6
```
AnkiDroid Version = 2.16alpha61

Android Version = 12

Manufacturer = samsung

Model = SM-N770F

Hardware = exynos9810

Webview User Agent = Mozilla/5.0 (Linux; Android 12; SM-N770F Build/SP1A.210812.016; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/100.0.4896.88 Mobile Safari/537.36

ACRA UUID = 452dd4db-5afb-487a-96d8-fcc7ccc857a8

Scheduler = std2

Crash Reports Enabled = false

DatabaseV2 Enabled = true
```

###### Research
*Enter an [x] character to confirm the points below:*

- [X] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [X] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [X] I have searched for similar existing issues here and on the user forum
- [X] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

### LINKED ISSUE #11060: [Bug] Option icons color changes on screen rotation
###### Reproduction Steps

1. Choose any deck
2. Click on the options menu on top right 
3. rotate your phone to landscape/portrait mode

https://user-images.githubusercontent.com/40427461/164959703-76e8e2e3-0bc0-4a90-8791-4ba246c4fd75.mp4

###### Expected Result
Color of the icons should not change

###### Actual Result
icons color changes

###### Debug info
AnkiDroid Version = 2.16alpha52

Android Version = 9

Manufacturer = asus

Model = ASUS_X01BDA

Hardware = qcom

Webview User Agent = Mozilla/5.0 (Linux; Android 9; ASUS_X01BDA Build/PKQ1; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/100.0.4896.127 Mobile Safari/537.36

ACRA UUID = e8337654-a2bd-4df5-850a-c4fe10d24fb3

Scheduler = std2

Crash Reports Enabled = false

DatabaseV2 Enabled = true

- [x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] I have searched for similar existing issues here and on the user forum
- [ ] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

### PR DESCRIPTION
**Pull request by dorrin-sot**
## Purpose / Description
Fixed Change in overflow menu icon colors if screen orientation changed

## Fixes
Fixes #11060 
Fixes #11017 

## How Has This Been Tested?
Tested on Emulator

https://user-images.githubusercontent.com/59933477/164968847-3578e565-d8b0-43ac-ab7c-b5634576bd37.mov

## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [x] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/res/values/theme_plain.xml [hunk 1]
```diff
@@ -38,7 +38,6 @@
         <!-- Note editor colors -->
         <item name="editTextBackgroundColor">#EBCCCCCC</item>
         <item name="toolbarBackgroundColor">#D6D7D7</item>
-        <item name="toolbarIconColor">@color/black</item>
```
Comment (krmanik) [inline AnkiDroid/src/main/res/values/theme_plain.xml:41]: This is not recommended fix because it makes note editor toolbar icon white.

</img>

**Review by krmanik — CHANGES_REQUESTED** [review]

File: AnkiDroid/src/main/res/values/styles.xml [hunk 2]
```diff
@@ -121,6 +121,7 @@
         <item name="android:textColorPrimary">?attr/actionBarPopupTextColor</item>
         <item name="android:textColorSecondary">?attr/actionBarPopupTextColor</item>
         <item name="android:colorBackground">?attr/popupBackgroundColor</item>
+        <item name="tint">@color/material_blue_600</item>
```
Comment (krmanik) [inline AnkiDroid/src/main/res/values/styles.xml:124]: It may changes color for other view in different activity. Have you checked it?

**Comment by dorrin-sot** [comment]
@krmanik Oh I found out that the flags menu was being set to blue too. Turns out, the issue was that if activity was restarted while the action menu was open, `onMenuOpened` would be called **before** `onCreateOptionsMenu` which would reset the colors.
I added a variable for optionsMenu being open, that would be saved in state and if at the end of `onCreateOptionsMenu` it was open, `shouldUseDefaultColor` would be called on menu items to set their color to blue

**Comment by mikehardy** [comment]
@BrayanDSO the above comment :point_up: might be related to the huge count of "menu bar is acting funny" issues perhaps?

**Comment by BrayanDSO** [comment]
I pulled this, there is only one issue with the menu items changing their colors 👇


#11017, #10697 and #11060 (same as #10732) are the "App bar items color issues". Only the `Undo button` wasn't solved since it's the same as the above.

I think the path taken by @dorrin-sot is right, only the above issue needs to be fixed.

Some other of the app bar issues may be related, but the connection is less clear, so needs investigation.

**Comment by david-allison** [comment]
Can we add a `@NeedsTest` or two around this area? 

Menu items + colors are new in 2.16, and seem to be the cause of a fair few issues

**Comment by mikehardy** [comment]
I think per @BrayanDSO investigation (maybe @dorrin-sot as well?) that it was a commit factoring out a chunk of abstract card viewer to reviewer and some order of operations thing that is causing a lot of issues actually...there is some interaction of invalidating items and setting them up again that's not working. It's documented on the related issues at least and they are cross-linked

**Comment by david-allison** [comment]
> I think per @BrayanDSO investigation (maybe @dorrin-sot as well?) that it was a commit factoring out a chunk of abstract card viewer to reviewer and some order of operations thing that is causing a lot of issues actually...there is some interaction of invalidating items and setting them up again that's not working. It's documented on the related issues at least and they are cross-linked

I thought that was only flags appearing. If it's more issues, then I wasn't aware, and it's on me to fix.

**Comment by BrayanDSO** [comment]
I guess that commit only affected the flags/marked since IIRC it only changed them. But overall, yes, I believe too that there is a problem with the order of operations of the action bar (and overflow)

**Comment by dorrin-sot** [comment]
> I pulled this, there is only one issue with the menu items changing their colors 👇 

Sorry I don't understand. Do you mean the part that non-overflow icons are also changing color? If so, I did try to research how to check if an icon ended up outside the overflow menu or not (if it was "showAsAction"="ifRoom") but couldn't find anything 😕. Would appreciate any tips and/or tricks on how to check and exclude them from getting colored 👀

**Comment by dorrin-sot** [comment]
BTW while testing this I found that if Reviewer is opened while on Landscape mode and then rotated, the icons that were marked as `showAsAction="ifRoom"` won't get back in Overflow menu and therefore take alot of space.

https://user-images.githubusercontent.com/59933477/165091031-da7b7ac8-90e0-4339-8025-551dbba76e88.mov

**Pull request merged**